TAO/orbsvcs/orbsvcs/Event/ECG_Mcast_EH.cpp

```diff
@@ -197,6 +200,7 @@ int
 TAO_ECG_Mcast_EH::delete_unwanted_subscriptions (
     Address_Set& multicast_addresses)
 {
+  ACE_GUARD_RETURN (TAO_SYNCH_MUTEX, ace_mon, this->lock_, -1);
```

Comment:
Hi Olli,
Can you explain your motivation for this change? It looks like you've seen situations where handle input and delete_unwanted_subscriptions have raced change the state of th subscription array. Is that correct?

If so, I first suggest editing the title of this pull request to be more specific. Second, Johnny is correct, the locks shouldn't be held  during invocations that might be blocking. I have more to say, but let me know if I am missing the point first.